### ISSUE TITLE
Tasks to support CRAN

### ISSUE THREAD

**Issue by hturner** [r-devel/r-project-sprint-2023#55]
### Discussed in https://github.com/r-devel/r-project-sprint-2023/discussions/54

<div type='discussions-op-text'>

<sup>Originally posted by **hturner** August 24, 2023</sup>
Uwe Ligges has some ideas for small tasks that volunteers might do to support the CRAN Team. This might include tasks such as
* Improving the information in notes/warnings/errors to give package authors more hints about what has to be changed
* Improving the processing of incoming packages to CRAN</div>

**Comment by llrs** [comment]
While I do not attend the sprint I'm interested in helping the CRAN Team. 5 minutes ago someone asked if there was a CRAN Task View for Package Development.

**Comment by hturner** [comment]
CRAN Task Views are a community initiative and moved to GitHub for better community engagement a couple of years ago, see: https://github.com/cran-task-views/ctv.

**Comment by llrs** [comment]
Yes, sorry I just wanted to point out that people are somewhat lost on how to best comply with CRAN policy. (There is a [new proposal](https://github.com/cran-task-views/ctv/issues/56))

**Comment by agila5** [comment]
Hi! I would also like to help on this issue during the sprint :)

**Comment by malcolmbarrett** [comment]
I missed the issue version of this, but I'm interested in the first task

**Comment by agila5** [comment]
We created a small and simple patch on point 1 and submitted it to Uwe. Happy to add more details during the wrapup session!

**Comment by mmaechler** [comment]
Thank you for the good idea and the patch.

To your 2nd part: No, that is not a typo, and your change would (almost surely) introduce a bug by doing something slightly differently than intended.

**Comment by llrs** [comment]
Thanks Martin for checking my first patch. 

<details>
<summary>Thanks, now I see it. I didn't check the context carefully enough, sorry.</summary>

Now I see that `Sys.setenv1` is defined in src/library/tools/R/utils.R with the following comment: "Sys.setenv() *one* variable unless it's set (to non-empty) already - export/move to base?". I didn't found it used anywhere else but I found some code that seems to be used for the same purpose on some lines above:

```r
prev <- Sys.getenv("_R_CHECK_SCREEN_DEVICE_", NA_character_)
if(is.na(prev)) Sys.setenv("_R_CHECK_SCREEN_DEVICE_" = "stop")
``` 

But I don't understand enough to see if `.Internal` makes a difference there. 
</details>


Getting back to the [MIT](https://cran.r-project.org/web/licenses/MIT), [BSD2](https://cran.r-project.org/web/licenses/BSD_2_clause) and [BSD3](https://cran.r-project.org/web/licenses/BSD_3_clause) and the template. 
The idea was to check if the file LICENSE has all the appropriate fields for the license.
I don't feel confident to propose such a change without checking interactively before. 
I could fabricate some tests and add them to the source code. 

Would it be fine if I propose some tests for external or internal function of tools? 
My idea would be to have some DESCRIPTION and LICENSE files to check (I might need to create a folder as some functions depend on it). 
I hope this way it will be quicker an easier to catch the major problems before proposing a patch. 
Would they go under the `test/`  or the `src/library/tools/tests/` folder?

**Comment by aitap** [comment]
There are [ideas](https://stat.ethz.ch/pipermail/r-devel/2023-July/082740.html) for `_R_CHECK_DEPENDS_ONLY_` when `.Library` contains user packages due to being writeable, but a different implementation (an environment variable telling R to pretend certain packages don't exist) may be needed in the end for [PR18584](https://bugs.r-project.org/show_bug.cgi?id=18584).

Another [`R CMD check` message improvement](https://hypatia.math.ethz.ch/pipermail/r-devel/2023-October/082935.html) implemented after the Sprint.
